src/nn/Enums.js

```diff
@@ -1178,6 +1178,62 @@ export const OperationCode = {
    */
   TANH: 28,
 
+  /**
+   * BatchToSpace for N-dimensional tensors.
+   *
+   * This operation reshapes the batch dimension (dimension 0) into M + 1
+   * dimensions of shape block_shape + [batch], interleaves these blocks back
+   * into the grid defined by the spatial dimensions [1, ..., M], to obtain a
+   * result with the same rank as the input.
+   *
+   * This is the reverse of SpaceToBatch.
+   *
+   * Supported tensor {@link OperandCode}:
+   * * {@link TENSOR_FLOAT32}
+   * * {@link TENSOR_QUANT8_ASYMM}
+   *
+   * Supported tensor rank: 4, with "NHWC" or "NCHW" data layout.
+   * With the default data layout NHWC, the data is stored in the order of:
+   * [batch, height, width, channels]. Alternatively, the data layout could
+   * be NCHW, the data storage order of: [batch, channels, height, width].
+   *
+   * Inputs:
+   * * 0: An n-D tensor, specifying the tensor to be reshaped.
+   * * 1: A 1-D Tensor of {@link TENSOR_INT32}, the block sizes for each 
+   *      spatial dimension of the input tensor. All values must be >= 1.
+   * * 2: An optional {@link BOOL} scalar, default to false.
```

Comment:
This scalar is available since NNAPI level 29. Should it be defined as an INT32 if we remain this optional scalar?

Reply:
How about we introduce the `BOOL` type?

Reply:
I agree with introducing `BOOL`. It's required by new `CONV` op, or we cannot distinguish implicit/exlicit padding by determine the number of inputs.

Reply:
Got it.

Reply:
As discussion, we will handle the NCHW/NHWC in following PR.